src/crashlog.cpp

```diff
@@ -324,6 +324,25 @@ char *CrashLog::LogRecentNews(char *buffer, const char *last) const
 	return buffer;
 }
 
+/**
+ * Create a timestamped filename.
+ * @param filename      The begin where to write at.
+ * @param filename_last The last position in the buffer to write to.
+ * @param ext           The extension for the filename.
+ * @param with_dir      Wether to append directory in front of filename.
+ * @return the number of added characters.
+ */
+int CrashLog::CreateFileName(char *filename, const char *filename_last, const char *ext, bool with_dir) const
+{
+	static std::string crashname;
+
+	if (crashname.empty()) {
+		UTCTime::Format(filename, filename_last, "crash%Y%m%d%H%M%S");
```

Comment:
I wonder how likely it is for this to trigger an allocation that OOMs the crashlog when its being generated. I seem to remember we were fairly wary about memory allocation before at least the crash log and dump were written. Not sure how big a problem it is, because it's a very small amount of memory that gets allocated though.

Reply:
On POSIX platforms at least, strictly speaking you should not call malloc or similar functions from a signal handler (including the crash log handler) because these are not re-entrant or signal-safe. OOM is not really an issue because if it happens it's more likely that the OOM killer will kill the whole process outright anyway. In the vast majority of cases you can get away with using malloc, and indeed the backtrace call implementations in the Unix crash log handler call malloc.
On Windows both writing the crash log and showing the dialog with the crash log contents in will allocate vastly more memory than this, so if an OOM event occurs any crash log will never make it to the user.

This string content here is short enough that it'll probably fit into the inline buffer anyway.